src/foam/core/session/PersistentSessionDAO.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * @license
+ * Copyright 2025 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.core.session',
+  name: 'PersistentSessionDAO',
+  extends: 'foam.dao.ProxyDAO',
+  implements: [
+    'foam.core.COREService'
+  ],
+
+  documentation: `The key to 'persistent' sessions is the userId on the
+session. If a user makes a client call with an existing session id and
+a matching user id, then the system considers the user logged in.
+Sessions are NOT-frozen, for reasons discussed elsewhere.  The
+implication is updates are not persisted to the runtime journal because
+there is never a delta.
+This DAO maintains it's own copy of the session to calculate the delta
+and then makes an explicit journal put call to store the session with
+the user id.
+
+Additionally, anonymous also adds a complication.  When a user is logged
+out, the client will present anonymous views and the anonymous user
+uses the same session id as it is the same client connection. This DAO
+also ensures the session is not saved with the anonymous user id.
```

Comment:
Wouldn't it be safer to save the session with an anonymous user id?  If somebody logs out, then their session should revert to the anonymous userid, we should persist that, otherwise if the server restarts won't they get logged back in?